source/elens.f90

```diff
@@ -395,48 +854,224 @@ subroutine elens_postInput
+end subroutine elens_postLinopt
 
+! ================================================================================================ !
+!  Compute geometrical normalisation factor
+! ================================================================================================ !
+subroutine elens_setOptics(iElem, bAlpha, bBeta, bOrbit, bOrbitP, bDisp, bDispP)
+
+  use parpro,     only : nele
+  use mod_common, only : bez
+  use numerical_constants, only : zero, two
+  use mod_settings, only : st_debug
+
+  integer,          intent(in) :: iElem
+  real(kind=fPrec), intent(in) :: bAlpha(2)
+  real(kind=fPrec), intent(in) :: bBeta(2)
+  real(kind=fPrec), intent(in) :: bOrbit(2)
+  real(kind=fPrec), intent(in) :: bOrbitP(2)
+  real(kind=fPrec), intent(in) :: bDisp(2)
+  real(kind=fPrec), intent(in) :: bDispP(2)
+
+  if(iElem<1 .or. iElem>nele .or. ( elens_r1(ielens(iElem))>zero .and. &
+       elens_r2(ielens(iElem))>zero .and. elens_sig(ielens(iElem))>zero)) return
+
+  elens_nUpdates(ielens(iElem))=elens_nUpdates(ielens(iElem))+1
+
+  if (elens_emin(ielens(iElem))>zero) then ! betatron cut
+    select case (elens_iSet(ielens(iElem)))
+    case(1) ! min
+       if (elens_optVal(ielens(iElem)).eq.zero) then ! first time
+         elens_optVal(ielens(iElem))=min(bBeta(1),bBeta(2))
+       else   
+         elens_optVal(ielens(iElem))=min(bBeta(1),bBeta(2),elens_optVal(ielens(iElem)))
+       end if
+    case(2) ! max
+       elens_optVal(ielens(iElem))=max(bBeta(1),bBeta(2),elens_optVal(ielens(iElem)))
+    case(3) ! average
+       elens_optVal(ielens(iElem))=elens_optVal(ielens(iElem))+(bBeta(1)+bBeta(2))/two
+    case(4) ! quadratic average
+       elens_optVal(ielens(iElem))=sqrt((elens_optVal(ielens(iElem))*elens_optVal(ielens(iElem)))+(bBeta(1)**2+bBeta(2)**2)/two)
+    case(5) ! geometric average
+       elens_optVal(ielens(iElem))=sqrt((elens_optVal(ielens(iElem))*elens_optVal(ielens(iElem)))*(bBeta(1)*bBeta(2)))
+    end select ! case (elens_iSet(ielens(iElem)))
+  else ! momentum cut
+    select case (elens_iSet(ielens(iElem)))
+    case(1) ! min
+       if (elens_optVal(ielens(iElem)).eq.zero) then ! first time
+         elens_optVal(ielens(iElem))=min(abs(bDisp(1)),abs(bDisp(2)))
+       else   
+         elens_optVal(ielens(iElem))=min(abs(bDisp(1)),abs(bDisp(2)),elens_optVal(ielens(iElem)))
+       end if
+    case(2) ! max
+       elens_optVal(ielens(iElem))=max(abs(bDisp(1)),abs(bDisp(2)),elens_optVal(ielens(iElem)))
+    case(3) ! average
+       elens_optVal(ielens(iElem))=elens_optVal(ielens(iElem))+(abs(bDisp(1))+abs(bDisp(2)))/two
+    case(4) ! quadratic average
+       elens_optVal(ielens(iElem))=sqrt((elens_optVal(ielens(iElem))*elens_optVal(ielens(iElem)))+(bDisp(1)**2+bDisp(2)**2)/two)
+    case(5) ! geometric average
+       elens_optVal(ielens(iElem))=sqrt((elens_optVal(ielens(iElem))*elens_optVal(ielens(iElem)))*(abs(bDisp(1))*abs(bDisp(2))))
+    end select ! case (elens_iSet(ielens(iElem)))
+  end if
```

Comment:
Should these to select statements have a `case default`?

Reply:
no, since the concerned variable is initialised in `elens_parseInputLine`, always via a `select` statement, with the `default`